### PR TITLE
Implement Feret diameters on Mask

### PR DESCRIPTION
**Pull request by opatiny**
- fix(Roi): make map readonly
- refactor: move getAngle to own file
- feat: implement getFeret
- docs: document feret interfaces
- fix: getPolygonArea returned negative values
- fix: convex Hull
- wip: working on getFeret
- test: enhance feret test
- test(getFeret): all tests pass except one
- docs(demo): add getFeret demo
- test(getFeret): all tests are good

### CODE REVIEW

File: src/maskAnalysis/__tests__/getConvexHull.test.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
+import { toBeDeepCloseTo, toMatchCloseTo } from 'jest-matcher-deep-close-to';
+
 import { getConvexHull } from '../getConvexHull';
 
+expect.extend({ toBeDeepCloseTo, toMatchCloseTo });
```
Comment (targos) [inline src/maskAnalysis/__tests__/getConvexHull.test.ts:5]: This is should be done in `jest.setup.ts`.

File: src/maskAnalysis/__tests__/getFeret.test.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
+import { toBeDeepCloseTo, toMatchCloseTo } from 'jest-matcher-deep-close-to';
+
+expect.extend({ toBeDeepCloseTo, toMatchCloseTo });
```
Comment (targos) [inline src/maskAnalysis/__tests__/getFeret.test.ts:3]: ditto

File: src/maskAnalysis/__tests__/getFeret.test.ts [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+describe('getFeret', () => {
```
Comment (targos) [inline src/maskAnalysis/__tests__/getFeret.test.ts:5]: please remove the useless describe :)

File: src/maskAnalysis/__tests__/getConvexHull.test.ts (hunk 1 above)
Comment (opatiny) [reply src/maskAnalysis/__tests__/getConvexHull.test.ts:5]: I have the impression that if I import in `jest.setup.ts` I don't need to do it at all in the test files, is this correct?

Comment (targos) [reply src/maskAnalysis/__tests__/getConvexHull.test.ts]: yes

File: jest.setup.ts [hunk 4]
```diff
@@ -8,4 +8,6 @@ expect.extend({
   ...jestMatchers,
 });
 
+expect.extend({ toBeDeepCloseTo, toMatchCloseTo });
```
Comment (targos) [inline jest.setup.ts:11]: You can just update the existing extend above.

File: src/maskAnalysis/__tests__/getFeret.test.ts [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+          { column: 0, row: 0 },
+          { column: 0, row: 0 },
+        ],
+      },
+      aspectRatio: 1,
+    },
+    3,
+  );
+});
+test.skip('mask with only 1 pixel', () => {
```
Comment (targos) [inline src/maskAnalysis/__tests__/getFeret.test.ts:27]: either fix the test/implementation or use `test.failing`

File: jest.setup.ts (hunk 4 above)
Comment (opatiny) [reply jest.setup.ts:11]: Woops '^^

**Pull request merged**